scripts/lib/upgrade-zulip-stage-2

```diff
@@ -432,6 +439,19 @@ if not args.skip_puppet and IS_SERVER_UP:
         sys.exit(1)
 
 
+def run_hooks(kind: Literal["pre-deploy", "post-deploy"]) -> None:
+    # Updates to the above literal to add a new hook type should
+    # adjust puppet's `app_frontend_base.pp` as well.
+    path = f"/etc/zulip/hooks/{kind}.d"
+    if not os.path.exists(path):
+        return
+    for script_name in sorted(f for f in os.listdir(path) if f.endswith(".hook")):
+        subprocess.check_call(
+            [os.path.join(path, script_name), old_version, NEW_ZULIP_VERSION],
+            cwd=deploy_path,
+        )
```

Comment:
I feel like it might be a nice touch to print each hook name as it runs, either using `run()` or otherwise, but I'm not sure and happy to consider that as a follow-up if we feel it warranted.